tools/io.cpp

```diff
@@ -87,12 +361,39 @@ bool ReadBinaryFile(const char* filename, std::vector<uint32_t>* data) {
     fp = stdin;
   }
 
-  ReadFile(fp, data);
-  bool succeeded = WasFileCorrectlyRead<uint32_t>(fp, filename);
+  // Read into a char vector first.  If this is a hex stream, it needs to be
+  // processed as such.
+  std::vector<char> data_raw;
+  ReadFile(fp, &data_raw);
```

Comment:
I don't know what percentage increase this will be, but for a game that wants to optimize 50,000 or 100,000 shaders each with an individual call to the command line, this could add up.

Reply:
ReadFile reads in chunks of 1024*sizeof(T), so basically 4KB at a time before my change and 1KB after, I fixed that so the read size is always 4KB.

In terms of performance, I gave it a try. Calling `spirv-dis` on the same moderately-sized file 2000 times takes about 4.5s on my computer. This exaggerates the cost of the extra copy because `spirv-dis` doesn't do much compared with `spirv-opt`, and reading the same file means the OS cache is hot (i.e. less work elsewhere, so cost of copy is higher in comparison). I don't see anything outside of noise before or after the change.